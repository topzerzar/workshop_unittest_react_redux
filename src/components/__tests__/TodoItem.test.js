import React from 'react'
import { shallow } from 'enzyme'
import TodoItem from '../TodoItem'

// Describe what to test
describe('TodoItem', () => {
  let component
  beforeEach(() => {
    component = shallow(<TodoItem text="Not Complete" />)
  })
  // Test Scenario
  it('match its snapshot - not complete', () => {
    expect(component).toMatchSnapshot()
  })

  it('match its snapshot - complete', () => {
    component.setProps({ complete: true }, { text: 'Hello' })
    expect(component).toMatchSnapshot()
  })

  it('render correct structure', () => {
    expect(component.is('li')).toBe(true)
  })

  it('has complete class if receive complete props', () => {
    component.setProps({ complete: true, text: 'Hello' })
    expect(component.hasClass('complete')).toBe(true)
    expect(component.text()).toBe('Hello')
  })
})

