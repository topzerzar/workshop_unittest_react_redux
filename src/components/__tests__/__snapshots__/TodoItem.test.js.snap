// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem match its snapshot - complete 1`] = `
<li
  className="complete"
>
  Not Complete
</li>
`;

exports[`TodoItem match its snapshot - not complete 1`] = `
<li
  className=""
>
  Not Complete
</li>
`;
